src/components/Context/Context.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  useCallback,
+  useState,
+  useMemo,
+  createContext,
+  ReactNode,
+  useEffect,
+  useContext,
+} from 'react';
+
+import * as TodoMethods from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { TypeOfFilter } from '../../types/TypeOfFilters';
+import { Error } from '../../types/TypeOfErrors';
+
+const USER_ID = 12049;
+
+interface AppContextTodoType {
+  USER_ID: number,
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  filter: TypeOfFilter,
+  setFilter: React.Dispatch<React.SetStateAction<TypeOfFilter>>,
+  errors: Error | null,
+  setErrors: React.Dispatch<React.SetStateAction<Error | null>>,
+  tempTodo: Todo | null,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>,
+  isLoading: boolean,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  filteredTodos: Todo[],
+  deleteTodoHandler: (value: number) => void,
+  addTodoTitle: (todo: Omit<Todo, 'id'>) => Promise<void>,
+  deleteIds: number[],
+  setDeleteIds: (value: number[]) => void,
+}
+
+export const appContext = createContext<AppContextTodoType>({
+  USER_ID: 12049,
+  todos: [],
+  setTodos: () => { },
+  filter: TypeOfFilter.All,
+  setFilter: () => { },
+  errors: null,
+  setErrors: () => { },
+  tempTodo: null,
+  setTempTodo: () => { },
+  isLoading: false,
+  setIsLoading: () => { },
+  filteredTodos: [],
+  deleteTodoHandler: () => { },
+  addTodoTitle: async () => { },
+  deleteIds: [],
+  setDeleteIds: () => { },
+});
+
+export const useAppContext = () => useContext(appContext);
+
+type Props = {
+  children: ReactNode;
+};
+
+export const AppContextProvider: React.FC<Props> = (props) => {
+  const { children } = props;
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TypeOfFilter>(TypeOfFilter.All);
+  const [errors, setErrors] = useState<Error | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [deleteIds, setDeleteIds] = useState<number[]>([]);
+
+  const filteredTodos = useMemo(() => {
+    if (filter === TypeOfFilter.Active) {
+      return todos.filter(todo => !todo.completed);
+    }
+
+    if (filter === TypeOfFilter.Completed) {
+      return todos.filter(todo => todo.completed);
+    }
+
+    return todos;
+  }, [todos, filter]);
```

Comment:
You have already created the `filterTodos` helper, it could be applied here and not duplicate the same code